digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/csvDataImporter/MassTransitStopCsvImporter.scala

```diff
@@ -0,0 +1,438 @@
+    updateAsset(externalId, None, properties.filterNot(_.columnName == "external_id"), roadTypeLimitations, user)
+      .map(excludedRoadLinkType => ExcludedRow(affectedRows = excludedRoadLinkType.toString, csvRow = rowToString(row)))
+  }
+}
+
+class Creator(vvhClientImpl: VVHClient, roadLinkServiceImpl: RoadLinkService, eventBusImpl: DigiroadEventBus) extends CsvOperations {
+  override def withDynTransaction[T](f: => T): T = OracleDatabase.withDynTransaction(f)
+  override def withDynSession[T](f: => T): T = OracleDatabase.withDynSession(f)
+  override def roadLinkService: RoadLinkService = roadLinkServiceImpl
+  override def vvhClient: VVHClient = vvhClientImpl
+  override def eventBus: DigiroadEventBus = eventBusImpl
+
+  override val decisionFieldsMapping = coordinateMappings
+
+  override def mandatoryFieldsMapping: Map[String, String] = coordinateMappings ++ stopAdministratorProperty ++ multipleChoiceFieldMappings
+
+  def getDirection(roadLink: RoadLink): ParsedProperties = {
+      val direction = roadLink.trafficDirection  match {
+        case TrafficDirection.BothDirections => TrafficDirection.TowardsDigitizing
+        case _ => roadLink.trafficDirection
+      }
+
+    List(AssetProperty("vaikutussuunta", TrafficDirection.toSideCode(direction).value))
+  }
+
+  override def createOrUpdate(row: Map[String, String], roadTypeLimitations: Set[AdministrativeClass], user: User, properties: ParsedProperties): List[ExcludedRow] = {
+    println("Creating busStop")
+    val lon = getPropertyValue(properties, "maastokoordinaatti_x").asInstanceOf[BigDecimal].toDouble
+    val lat = getPropertyValue(properties, "maastokoordinaatti_y").asInstanceOf[BigDecimal].toDouble
+
+    val nearestRoadLink = getNearestRoadLink(lon, lat, user, roadTypeLimitations)
+
+    if(nearestRoadLink.isEmpty)
+      List(ExcludedRow(affectedRows = "RoadLink no longer available", csvRow = rowToString(row)))
+    else {
+      val roadLink = roadLinkService.enrichRoadLinksFromVVH(nearestRoadLink)
+
+      val prop = convertProperties(getDirection(roadLink.head) ++ properties)
+      val mValue = GeometryUtils.calculateLinearReferenceFromPoint(Point(lon, lat), roadLink.head.geometry)
+      val bearing = GeometryUtils.calculateBearing(roadLink.head.geometry, Some(mValue))
+      val point = GeometryUtils.calculatePointFromLinearReference(roadLink.head.geometry, mValue)
+      val asset = NewMassTransitStop(point.map(_.x).getOrElse(lon), point.map(_.y).getOrElse(lat), roadLink.head.linkId, bearing, prop)
+
+      massTransitStopService.checkDuplicates(asset) match {
+        case Some(existingAsset) =>
+          updateAsset(existingAsset.nationalId, None, properties, roadTypeLimitations, user)
+            .map(excludedRoadLinkType => ExcludedRow(affectedRows = excludedRoadLinkType.toString, csvRow = rowToString(row)))
+        case _ =>
+          massTransitStopService.createWithUpdateFloating(asset, user.username, roadLink.head)
```

Comment:
Why are we calling `createWithUpdateFloating` directly instead of `def create`? Now we are not publishing any changes.